hollow/src/main/java/com/netflix/hollow/core/read/engine/HollowBlobReader.java

```diff
@@ -189,13 +193,23 @@ private String readTypeStateSnapshot(DataInputStream is, HollowBlobHeader header
             if(!filter.doesIncludeType(schema.getName())) {
                 HollowSetTypeReadState.discardSnapshot(is, numShards);
             } else {
-                populateTypeStateSnapshot(is, new HollowSetTypeReadState(stateEngine, (HollowSetSchema)schema, numShards));
+                HollowSetSchema setSchema = (HollowSetSchema) schema;
+                // Filter out hash key if overridden by a hash code finder
+                if (hasDefinedHashCode(setSchema.getElementType())) {
```

Comment:
Good point.  This is a transitory solution to retain some compatibility until consumers can remove the dependency on `HollowObjectHashCodeFinder`.  We could change the approach.  For example, the producer filters to remove the hash keys before publishing (since the producer and consumers both require knowledge of the schema types).

Reply:
I updated the main section with an addition comment. This is likely rare since prior to this change a consumer is unlikely to implement `getTypesWithDefinedHashCodes` as it was only actionable for the producer.